lib/src/component/hooks.dart

```diff
@@ -0,0 +1,535 @@
+// Copyright 2020 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:over_react/over_react.dart';
+import 'package:react/hooks.dart' as react_hooks;
+
+/// Adds local state to a [uiFunction] component
+/// by returning a [StateHook] with [StateHook.value] initialized to [initialValue].
+///
+/// > __Note:__ If the [initialValue] is expensive to compute, [useStateLazy] should be used instead.
+///
+/// __Example__:
+///
+/// ```dart
+/// UiFactory<UiProps> UseStateExample = uiFunction(
```

Comment:
Same comnments as in examples around formatting and using generated function configs. I think here we can omit the props mixin declarations themselves, though, and just use them in the typing.